### ISSUE TITLE
ocaml version inside install_ppa function

### ISSUE THREAD

**Issue by gitoleg** [ocaml/ocaml-ci-scripts#233]
There is an `install_on_linux` function https://github.com/ocaml/ocaml-ci-scripts/blob/master/.travis-ocaml.sh#L84 
It sets env variables like `OCAML_VERSION` and `OCAML_FULL_VERSION`  _AFTER_ calling `apt-get install ocaml ocaml-base ...`, i.e. after call of `install_ppa`,  while in the previous version it was done _BEFORE_, so actually `apt-get install`  worked before with updated env, e.g. `full_apt_version ocaml` was expanded to `ocaml=4.02.3-2ppa4~trusty`, because `OCAML_VERSION` was set to `4.02.3`. 
In other words, a user-defined `OCAML_VERSION` is used for building now, not for tests only, while it wasn't so before last changes.
So a question - Is it correct? In our case, it breaks lot's of stuff, because we do build for many versions of `ocaml`.And previously it worked well just because the user-defined version was used for tests only, but not for building. 
Thanks!

**Comment by samoht** [comment]
Ha yes indeed, I have changed the order of `OCAML_VERSION` being set and installing the ppa while updating to opam2; Sorry about that, I'll revert these.
